marshal/claim.go

```diff
@@ -261,6 +263,13 @@ func (c *claim) teardown(releasePartition bool) bool {
 	// is reasonable. Held because of using offsetCurrent below.
 	c.lock.RLock()
 	defer c.lock.RUnlock()
+
+	// Advise the consumer that this claim is terminating, this is so that the consumer
+	// can release other claims if we've lost part of a topic
+	if c.consumer != nil {
+		go c.consumer.claimTerminated(c, releasePartition)
```

Comment:
It is not possible to synchronously release all claims on a topic anyway. That's not really supported in the protocol. The best you could get would be to try to batch produce and assert that everything produced or not, but even then Kafka doesn't guarantee that it will only ever commit whole batches. Since the underlying protocl is asynchronous it's fine for this to be asynchronous, I believe.

How does this affect your cleanup?